Changes for online offices

ngulden opened the pull request:
The virtual app appliances Collabora with Nextcloud and ONLYOFFICE with Nextcloud, the app needs the following update. When the appliance is setup, a online office app is installed first followed by Nextcloud. The goal is to have a system ready where the system administrator does not have to configure anything in order to have one of the online offices working together with Nextcloud.

This pull requests requires https://github.com/nextcloud/univention-app/pull/63 because it assumes that the respective Nextcloud apps for either online office is already installed in the Docker Image.

ngulden committed:
Add other appliances to appliance detection

ngulden committed:
Add auto detection for online offices

When either Collabora Online, Collabora CODE or ONLYOFFICE are installed
on a UCS system when Nextcloud is going to be installed, the respective
apps in Nextcloud are activated. If no URL to the services are
configured, they are set to the given system.

If no online office is installed, the join script will deactivate the
respective app in Nextcloud.

ngulden commented:
@blizzz : I could not determine how to make a check for the installation status of those Nextcloud apps first. Maybe you have a hint and I add it.

blizzz commented on inst:
```diff
@@ -338,4 +341,38 @@ nextcloud_curl() {
+    if [ "$(ucr get appcenter/apps/collabora/status)" = "installed" ] || [ "$(ucr get appcenter/apps/collabora-online/status)" = "installed" ] ; then
+        univention-app shell nextcloud sudo -u www-data /var/www/html/occ app:enable richdocuments
+        occCode=$?
+        if [ "$(univention-app shell nextcloud sudo -u www-data /var/www/html/occ config:app:get richdocuments wopi_url)" = "" ] && [ ${occCode} -eq 0 ] ; then
```
micro-optimization: checking the result code first would save the request into the docker container if the app was not enabled.

ngulden replied:
I updated the check to reflect your feedback. Thank you.

blizzz commented on inst:
```diff
@@ -338,4 +341,38 @@ nextcloud_curl() {
+            univention-app shell nextcloud sudo -u www-data /var/www/html/occ config:app:set richdocuments wopi_url --value="https://$FQDN/"
+        fi
+    else
+        univention-app shell nextcloud sudo -u www-data /var/www/html/occ app:disable richdocuments
```
I would not disable it to avoid false positives. There is a chance that an external office server is in use.

ngulden replied:
The use case here is that Nextcloud is in the process of being installed. If an external office server should be used, it has to be configured manually directly in Nextcloud.

If a document server is already present on the same system, this configuration is done automatically.

If no document server is used locally, it is deactivated. If not, the app is there and it offers the user to open office files. This won't work without a document server and a "Not found" (app uninstalled) or "Service unavailable" (app stopped) error is shown to the user in Nextcloud.

I believe it is better to let the user manually reactivate the office app, if the document server is remote, rather than confronting the user with an ugly error message.

blizzz replied:
Unless something has changed, it is possible that an update routine is being run (we check it an set a flag with `IS_UPDATE`). In that case, the admin should not need to re-enable it. And otherwise it is not enabled either.

ngulden replied:
I fear I don't understand your point enough to add appropriate code.

blizzz replied:
Let me rephrase:

1. The default state is that the app is disabled
2. This routine will be entered on Nextcloud app update, too, not only installation
3. → disabling is not necessary, but could disable an intentionally enabled app → bad UX
=> let's just drop this line of code

ngulden replied:
All right. I just made a new commit and dropped the disable lines.

ngulden committed:
Change order for exit code check

ngulden committed:
Drop explicit disabling of online office apps

When the apps are installed in Nextcloud in the docker image, they are
disabled by default. There is no need to disable them. Running the inst
script upon update could disable the online office apps unintentionally.
This is avoided now.
